warp/Network/Wai/Handler/Warp/Response.hs

```diff
@@ -351,6 +285,76 @@ sendRsp conn _ _ _ _ (RspRaw withApp src tickle) = do
     send bs = connSendAll conn bs >> tickle
 
 ----------------------------------------------------------------
+
+-- Sophisticated WAI applications.
+-- We respect s0. s0 MUST be a proper value of either 200 or 206.
```

Comment:
Why can't we have, for example, a 404 response that streams part of a file? I can't picture actually needing it, but this seems like a strange limitation.